src/core/src/validation_util.cpp

```diff
@@ -1229,22 +1249,31 @@ HostTensorPtr evaluate_bound(const Output<Node>& output, bool is_upper) {
                 TensorLabelVector output_labels(outputs.size());
 
```

Comment:
I would use dedicated `const size_t max_elements_limit = 10;` for `are_equal` and later checks during invalidation just to have this value in a single place.